pkg/detectors/uri/uri.go

```diff
@@ -23,7 +21,9 @@ type Scanner struct {
 var _ detectors.Detector = (*Scanner)(nil)
 
 var (
-	keyPat = regexp.MustCompile(`\b[a-zA-Z]{1,10}:?\/\/[-.%\w{}]{1,50}:([-.%\S]{3,50})@[-.%\w\/:]+\b`)
+	keyPat = regexp.MustCompile(`\b(?:http(?:s)?:)?\/\/[\S]{3,50}:([\S]{3,50})@[-.%\w\/:]+\b`)
```

Comment:
suggestion: simplify this regex and unwrap the extra capture group?

ex: `\b(?:https?:)?//\S{3,50}:(\S{3,50})@[-.%\w/:]+\b`